manifests/service.pp

```diff
@@ -3,11 +3,11 @@
 #
 # @api private
 class apache::service (
-  $service_name           = $apache::params::service_name,
-  Boolean $service_enable = true,
-  $service_ensure         = 'running',
-  Boolean $service_manage = true,
-  $service_restart        = undef
+  String $service_name                    = $apache::params::service_name,
+  Boolean $service_enable                 = true,
+  Variant[Boolean,String] $service_ensure = 'running',
```

Comment:
What are the possible values here? Should it just be a string? Rather than the variant?

Reply:
service_ensure can be set too true or false in the code. There are examples of each in the tests

Reply:
https://github.com/puppetlabs/puppetlabs-apache/blob/babc21bb524d2bae88066fe8abeeb7e77ace960a/manifests/service.pp#L16

Oh i see here... that doesn't seem right to me but it's a conversation that doesn't fall in the scope of this PR so I'm happy. That goes for the other questions too!